src/codec/lzma/encoder.rs

```diff
@@ -0,0 +1,96 @@
+use crate::{codec::Encode, util::PartialBuffer};
+
+use std::fmt::{Debug, Formatter, Result as FmtResult};
+use std::io::Result;
+use xz2::stream::{Action, Check, Status, Stream};
+
+pub struct LzmaEncoder {
+    stream: Stream,
+}
+
+impl Debug for LzmaEncoder {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        write!(f, "LzmaEncoder")
+    }
+}
+
+impl LzmaEncoder {
+    pub fn new(level: u32) -> Self {
+        Self {
+            stream: Stream::new_easy_encoder(level, Check::Crc64).unwrap(),
```

Comment:
This and the decoder appear to be using [the XZ file format](https://tukaani.org/xz/xz-file-format.txt) rather than raw LZMA. Should they be called `XzDecoder`/`XzEncoder` to indicate that, or did you mean to actually have raw LZMA here? (Potentially it would make sense to support all of LZMA, LZMA2 and XZ formats as separate encoder/decoder pairs).

Reply:
You are right, they should probably be called `XzDecoder`/`XzEncoder`. I thought `lzma` and `xz` could be used interchangeably, but i was wrong. Then again I am not sure what an LZMA2 decoder/encoder would do, as the en-/decoding process seems to only make sense in the context of a specific file format? So we could do `LzmaEncoder` for `.lzma` files and `XzEncoder` for `.xz` files?
Or am I still misunderstanding how the LZMA/LZMA2 interact with the various file formats?

Reply:
It would be possible to use LZMA2 as part of some other kind of data encoding, e.g. the `xz` format itself wraps an LZMA2 data stream. But by the looks of it `xz2` doesn't support encoding directly to LZMA2, `new_easy_encoder` gives you an `xz` encoder, `new_lzma_encoder` gives you an `lzma` encoder and `new_stream_encoder` gives you a customised `xz` encoder.

So for now adding separate `LzmaEncoder` and `XzEncoder` would be great (under separate `xz` and `lzma` features, probably sharing a base `Xz2Encoder`, similar to how `zlib` and `deflate` share most of their implementation).

Reply:
Ok, I started with what you suggested. Will have to add tests for the new LzmaEncoder/Decoder later today or tomorrorw.